### ISSUE TITLE
Overlapping overload not reported when parameter type includes overlapping unions

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6825]
This code should produce an "overlapping override" error, but it doesn't. Mypy currently flags this condition.

```python
class A:
    @overload
    def select(self, p1: int | set[int], /) -> str:
        ...

    @overload
    def select(self, p1: int | list[int], /) -> int:
        ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
